pyproject.toml

```diff
@@ -12,6 +12,7 @@ python = "^3.12"
 numpy = "^2.0.0"
 scipy = "^1.14.0"
 matplotlib = "^3.9.1"
+scikit-learn = "^1.5.1"
```

Comment:
Also it seems we don't use this dependency so we can remove it.

`poetry remove scikit-learn` I guess.